rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+                NamespaceToRenameToDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          description: Not Acceptable - Unsupported Operation
+        409:
+          description: Conflict - The target table identifier to rename to already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              example:
+                $ref: '#/components/examples/TableAlreadyExistsError'
+
+components:
+  #######################################################
+  # Common Parameter Definitions Used In Several Routes #
+  #######################################################
+  parameters:
+    namespace:
+      name: namespace
+      in: path
+      required: true
+      description:
+        A namespace identifier as a single string.
+        Multipart namespace parts should be separated by the null byte, %00.
+      schema:
+        type: string
+      examples:
+        singlepart_namespace:
+          value: "accounting"
+        multipart_namespace:
+          value: "accounting%00tax"
+
+    table:
+      name: table
+      in: path
+      description: A table name
+      required: true
+      schema:
+        type: string
+      example: "sales"
+
+    parent:
+      name: parent
+      in: query
+      description:
+        Optional parent namespace under which to list namespaces.
+        If parent is a multipart namespace, the parts must be separated by the null byte, %00.
+        When empty, list top-level namespaces.
+      required: false
+      schema:
+        type: string
+      example: "accounting%00tax"
+
+  ##############################
+  # Application Schema Objects #
+  ##############################
+  schemas:
+
+    ResponseDataObject:
+      type: object
+      description: JSON data payload returned in a successful response body
+      properties:
+        data:
+          type: object
+          description: Wrapper for the response of a successful request
+      example: { "data": { ... } }
+
+    ResponseErrorObject:
+      type: object
+      description: JSON error payload returned in a response with further details on the error
+      required:
+        - message
+        - type
+        - code
+      properties:
+        message:
+          type: string
+          description: Human-readable error message
+        type:
+          type: string
+          description: Internal type definition of the error
+          example: NoSuchNamespaceException
+        code:
+          type: integer
+          minimum: 400
+          maximum: 600
+          description: HTTP response code
+          example: 404
+
+    CatalogConfiguration:
+      type: object
+      description: Server-provided configuration for the catalog.
+      required:
+        - properties
+        - defaults
+      properties:
+        overrrides:
+          type: object
+          nullable: false
```

Comment:
Are these `nullable` settings needed? The property is already required and has a type.

Reply:
Technically, `nullable` signifies whether or not `null` is an allowed value (outside of whether or not the type of the value, e.g. String, would allow null in general).

But here they're just left over artifacts from before I had them under `required`. I'll remove the `nullable`.

Reply:
Removed.